lib/grell/crawler.rb

```diff
@@ -51,12 +51,21 @@ def crawl(site, block)
       Grell.logger.info "Visiting #{site.url}, visited_links: #{@collection.visited_pages.size}, discovered #{@collection.discovered_pages.size}"
       site.navigate
       filter!(site.links)
-
-      if block #The user of this block can send us a :retry to retry accessing the page
-        while block.call(site) == :retry
-          Grell.logger.info "Retrying our visit to #{site.url}"
-          site.navigate
-          filter!(site.links)
+      add_redirect_url(site)
+
+      if block # The user of this block can send us a :retry to retry accessing the page
+        begin
+          while block.call(site) == :retry
+            Grell.logger.info "Retrying our visit to #{site.url}"
+            site.navigate
+            filter!(site.links)
+            add_redirect_url(site)
+          end
+        rescue Capybara::Poltergeist::BrowserError, Capybara::Poltergeist::DeadClient,
```

Comment:
true, I forgot about that. Then maybe all that should go into a method so it is clear:
```
def ...
  block.call(site)
rescue Capybara::Poltergeist::BrowserError, Capybara::Poltergeist::DeadClient,
               Capybara::Poltergeist::JavascriptError, Capybara::Poltergeist::StatusFailError,
               Capybara::Poltergeist::TimeoutError, Errno::ECONNRESET, URI::InvalidURIError => e
  site.unavailable_page(404, e)
  return
end
```